### PR TITLE
Remove pytest fixture and fix test_tag/s method

### PR DESCRIPTION
**Pull request by stisac**
*Issue #, if available:* https://sim.amazon.com/issues/AML-106623

*Description of changes:* Removed pytest fixture in tests/integ/sagemaker/lineage/conftest.py which was overlapping with conftest.py in main and suppressing endpoint param passed via command line.

Also added a condition in test_tag and test_tags method to assert tags only if they aren't provided via command line

*Testing done:*

tox -e py39 -- tests/integ/sagemaker/lineage --sagemaker-client-config {\"endpoint_url\":\"https://api.sagemaker.beta.us-west-2.ml-platform.aws.a2z.com\"}

tests/integ/sagemaker/lineage/test_action.py .......                                                                                                                                                                                [ 25%]
tests/integ/sagemaker/lineage/test_artifact.py ........                                                                                                                                                                             [ 53%]
tests/integ/sagemaker/lineage/test_association.py sss                                                                                                                                                                               [ 64%]
tests/integ/sagemaker/lineage/test_context.py .......                                                                                                                                                                               [ 89%]
tests/integ/sagemaker/lineage/test_dataset_artifact.py .                                                                                                                                                                            [ 92%]
tests/integ/sagemaker/lineage/test_endpoint_context.py .                                                                                                                                                                            [ 96%]
tests/integ/sagemaker/lineage/test_model_artifact.py .                                                                                                                                                                              [100%]

### CODE REVIEW

**Comment by stisac** [comment]
The failure from 'sagemaker-python-sdk-pr' tests are resulting from tests/integ/test_multidatamodel.py and is unrelated to the changes made in this pull-request.

File: tests/integ/sagemaker/lineage/test_action.py [hunk 1]
```diff
@@ -90,8 +90,10 @@ def test_tag(action_obj, sagemaker_session):
         )["Tags"]
         if actual_tags:
             break
-    assert len(actual_tags) == 1
-    assert actual_tags[0] == tag
+
+    if len(actual_tags) == 1:
```
Comment (danabens) [inline tests/integ/sagemaker/lineage/test_action.py:94]: This doesn't make any sense - if the assertion is wrong change the assertion

File: tests/integ/sagemaker/lineage/test_action.py [hunk 2]
```diff
@@ -104,5 +106,7 @@ def test_tags(action_obj, sagemaker_session):
         )["Tags"]
         if actual_tags:
             break
-    assert len(actual_tags) == 1
-    assert actual_tags == tags
+
+    if len(actual_tags) == 1:
```
Comment (danabens) [inline tests/integ/sagemaker/lineage/test_action.py:110]: remove - why is this assertion failing? find the root cause

File: tests/integ/sagemaker/lineage/test_artifact.py [hunk 3]
```diff
@@ -99,8 +99,10 @@ def test_tag(artifact_obj, sagemaker_session):
         )["Tags"]
         if actual_tags:
             break
-    assert len(actual_tags) == 1
-    assert actual_tags[0] == tag
+
+    if len(actual_tags) == 1:
```
Comment (danabens) [inline tests/integ/sagemaker/lineage/test_artifact.py:103]: ditto

File: tests/integ/sagemaker/lineage/test_artifact.py [hunk 4]
```diff
@@ -113,5 +115,7 @@ def test_tags(artifact_obj, sagemaker_session):
         )["Tags"]
         if actual_tags:
             break
-    assert len(actual_tags) == 1
-    assert actual_tags == tags
+
+    if len(actual_tags) == 1:
```
Comment (danabens) [inline tests/integ/sagemaker/lineage/test_artifact.py:119]: ditto

File: tests/integ/sagemaker/lineage/test_context.py [hunk 5]
```diff
@@ -88,8 +88,10 @@ def test_tag(context_obj, sagemaker_session):
         )["Tags"]
         if actual_tags:
             break
-    assert len(actual_tags) == 1
-    assert actual_tags[0] == tag
+
+    if len(actual_tags) == 1:
```
Comment (danabens) [inline tests/integ/sagemaker/lineage/test_context.py:92]: ditto

File: tests/integ/sagemaker/lineage/test_context.py [hunk 6]
```diff
@@ -102,5 +104,7 @@ def test_tags(context_obj, sagemaker_session):
         )["Tags"]
         if actual_tags:
             break
-    assert len(actual_tags) == 1
-    assert actual_tags == tags
+
+    if len(actual_tags) == 1:
```
Comment (danabens) [inline tests/integ/sagemaker/lineage/test_context.py:108]: ditto

File: tests/integ/sagemaker/lineage/test_action.py (hunk 1 above)
Comment (stisac) [reply tests/integ/sagemaker/lineage/test_action.py:94]: The assertion is failing because the length of the actual_tags is now 2.
The tag is assigned with tag = {"Key": "foo", "Value": "bar"}

When we pass in the below argument for running integ tests
--sagemaker-client-config {"endpoint_url":"https://api.sagemaker.beta.us-west-2.ml-platform.aws.a2z.com\"}

The length of the actual_tags will become 2 as per function logic in test_tag or test_tags method.

The value in actual_tags is below.
([{'Key': 'foo', 'Value': 'bar'}, {'Key': 'aws:tag:domain', 'Value': 'beta'}])

If we don't pass the sagemaker-client-config and just run the command, it will be hitting prod endpoint directly.
And since addition of sagemaker-client-config argument is optional, I added the if condition.

Comment (danabens) [reply tests/integ/sagemaker/lineage/test_action.py:94]: we still want to make this assertion in beta so remove the `if` statement, change to `assert  len(actual_tags) > 0`, add a comment on this edge case

Comment (stisac) [reply tests/integ/sagemaker/lineage/test_action.py:94]: Sure thing.

File: tests/integ/sagemaker/lineage/test_action.py (hunk 2 above)
Comment (stisac) [reply tests/integ/sagemaker/lineage/test_action.py:110]: RC described above.. Issue addressed

File: tests/integ/sagemaker/lineage/test_artifact.py (hunk 3 above)
Comment (stisac) [reply tests/integ/sagemaker/lineage/test_artifact.py:103]: RC described above.. Issue addressed

File: tests/integ/sagemaker/lineage/test_artifact.py (hunk 4 above)
Comment (stisac) [reply tests/integ/sagemaker/lineage/test_artifact.py:119]: RC described above.. Issue addressed

File: tests/integ/sagemaker/lineage/test_context.py (hunk 5 above)
Comment (stisac) [reply tests/integ/sagemaker/lineage/test_context.py:92]: RC described above.. Issue addressed

File: tests/integ/sagemaker/lineage/test_context.py (hunk 6 above)
Comment (stisac) [reply tests/integ/sagemaker/lineage/test_context.py:108]: RC described above.. Issue addressed

**Review by danabens — APPROVED** [review]

**Comment by stisac** [comment]
The failure from 'sagemaker-python-sdk-pr' tests are resulting from tests/integ/test_multidatamodel.py and is unrelated to the changes made in this pull-request.

**Review by ahsan-z-khan — APPROVED** [review]
